overlord/devicestate/handlers_install.go

```diff
@@ -131,7 +134,13 @@ func (m *DeviceManager) doSetupRunSystem(t *state.Task, _ *tomb.Tomb) error {
 		bopts.Model = deviceCtx.Model()
 		bopts.SystemLabel = modeEnv.RecoverySystem
 
-		installObserver = boot.TrustedAssetsInstallObserverForModel(deviceCtx.Model())
+		trustedInstallObserver, err = boot.TrustedAssetsInstallObserverForModel(deviceCtx.Model())
+		if err != nil && err != boot.ErrObserverNotApplicable {
```

Comment:
maybe a comment explaining that we could have boot.ErrObserverNotApplicable in unencrypted cases would be useful